feature/gribi/mpls_in_udp/otg_tests/mpls_over_udp_tunnel_hashing_test/mpls_over_udp_tunnel_hashing_test.go

```diff
@@ -0,0 +1,935 @@
+	}
+
+	// Allow time for configuration to be applied
+	time.Sleep(10 * time.Second)
+
+}
+
+// configureBaseconfig configures network instances and forwarding policy on the DUT
+func configureBaseconfig(t *testing.T, dut *ondatra.DUTDevice) {
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
+	pf := getPbrPolicy(dut, clusterPolicy)
+	gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).PolicyForwarding().Config(), pf)
+}
+
+// applyForwardingPolicy applies the VRF selection policy to the ingress interface
+func applyForwardingPolicy(t *testing.T, dut *ondatra.DUTDevice, ingressPort string) {
+	d := &oc.Root{}
+	interfaceID := ingressPort
+	if deviations.InterfaceRefInterfaceIDFormat(dut) {
+		interfaceID = ingressPort + ".0"
+	}
+
+	pfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).PolicyForwarding().Interface(interfaceID)
+	pfCfg := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreatePolicyForwarding().GetOrCreateInterface(interfaceID)
+	pfCfg.ApplyVrfSelectionPolicy = ygot.String(clusterPolicy)
+	pfCfg.GetOrCreateInterfaceRef().Interface = ygot.String(ingressPort)
+	pfCfg.GetOrCreateInterfaceRef().Subinterface = ygot.Uint32(0)
+	gnmi.Replace(t, dut, pfPath.Config(), pfCfg)
+}
+
+// getPbrPolicy creates policy-based routing configuration for VRF selection
+func getPbrPolicy(dut *ondatra.DUTDevice, name string) *oc.NetworkInstance_PolicyForwarding {
+	d := &oc.Root{}
+	ni := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	pf := ni.GetOrCreatePolicyForwarding()
+	p := pf.GetOrCreatePolicy(name)
+	p.SetType(oc.Policy_Type_VRF_SELECTION_POLICY)
+
+	for _, pRule := range getPbrRules(dut) {
+		r := p.GetOrCreateRule(seqIDOffset(dut, pRule.sequence))
+
+		if deviations.PfRequireMatchDefaultRule(dut) {
+			if pRule.etherType != nil {
+				r.GetOrCreateL2().Ethertype = pRule.etherType
+			}
+		}
+
+		if pRule.encapVrf != "" {
+			r.GetOrCreateAction().SetNetworkInstance(pRule.encapVrf)
+		}
+	}
+	return pf
+}
+
+// getPbrRules returns policy-based routing rules for VRF selection
+func getPbrRules(dut *ondatra.DUTDevice) []pbrRule {
+	vrfDefault := deviations.DefaultNetworkInstance(dut)
+
+	if deviations.PfRequireMatchDefaultRule(dut) {
+		return []pbrRule{
+			{
+				sequence:  17,
+				etherType: ethertypeIPv4,
+				encapVrf:  vrfDefault,
+			},
+			{
+				sequence:  18,
+				etherType: ethertypeIPv6,
+				encapVrf:  vrfDefault,
+			},
+		}
+	}
+	return []pbrRule{
+		{
+			sequence: 17,
+			encapVrf: vrfDefault,
+		},
+	}
+}
+
+// seqIDOffset returns sequence ID with base offset to ensure proper ordering
+func seqIDOffset(dut *ondatra.DUTDevice, i uint32) uint32 {
+	if deviations.PfRequireSequentialOrderPbrRules(dut) {
+		return i + seqIDBase
+	}
+	return i
+}
+
+// staticARPWithMagicUniversalIP configures static ARP with magic universal IP
+func staticARPWithMagicUniversalIP(t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
move this to cfgplugins.  Please define parameters for the dut ports and next-hops so it will be reusable.  

I suggest to place it in `internal/cfgplugins/interface.go` with a function named: `StaticARPWithMagicUniversalIP`